pkg/build/generator/generator.go

```diff
@@ -93,6 +99,23 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 	if err != nil {
 		return nil, err
 	}
+
+	// Fetch the secrets associated with the BuilderServiceAccount.
+	// These secrets will be used for push/pull authentication
+	// FIXME: Add support for SourceSecret
+	if sa, err := g.ServiceAccounts.ServiceAccounts(bc.Namespace).Get(bootstrappolicy.BuilderServiceAccountName); err != nil {
+		glog.V(2).Infof("Unable to fetch builder service account for %s: %v", bc.Namespace, err)
+	} else {
+		for _, ref := range sa.Secrets {
+			secret, err := g.Secrets.Secrets(bc.Namespace).Get(ref.Name)
+			if err != nil {
+				glog.V(5).Infof("Unable to fetch builder secret %s/%s: %v", bc.Namespace, ref.Name, err)
+				continue
+			}
+			g.builderSecrets = append(g.builderSecrets, *secret)
```

Comment:
@deads2k I added argument to this function that you can use to specify the list of secrets you want to use, so when you're PR lands in upstream, we can add `imagePullSecrets` and use that instead of `builderSecrets`, PTAL.